### PR TITLE
Add fit option to scale

### PR DESCRIPTION
**Pull request by yuyu2172**

### CODE REVIEW

**Comment by rezoo** [comment]
Why did you decide to define ``size`` option as ``int``? Is it better to accept and integer and a tuple of ints?

**Comment by yuyu2172** [comment]
Sorry I am confused. What do you mean by tuple of ints?
Currently, the `size` indicates the length of the shorted edge.

File: chainercv/transforms/image/scale.py [hunk 1]
```diff
@@ -21,9 +28,14 @@ def scale(img, size):
     if (W <= H and W == size) or (H <= W and H == size):
         return img
```
Comment (Hakuyume) [inline chainercv/transforms/image/scale.py]: If `fit_short` is `False`, this `return` statement causes an undesired behaviour.

File: tests/transforms_tests/image_tests/test_scale.py [hunk 2]
```diff
@@ -6,19 +6,23 @@
...
+    {'in_shape': (3, 24, 16), 'size': 8,
+     'fit_short': True, 'out_shape': (3, 12, 8)},
+    {'in_shape': (3, 16, 24), 'size': 8,
+     'fit_short': True, 'out_shape': (3, 8, 12)},
+    {'in_shape': (3, 24, 16), 'size': 36,
+     'fit_short': False, 'out_shape': (3, 36, 24)},
+    {'in_shape': (3, 16, 24), 'size': 36,
+     'fit_short': False, 'out_shape': (3, 24, 36)}
+)
 class TestScale(unittest.TestCase):
```
Comment (Hakuyume) [inline tests/transforms_tests/image_tests/test_scale.py]: Cound you add more test cases that `size == H` and `size == W`?

**Comment by yuyu2172** [comment]
Thanks. I fixed the code.

**Comment by Hakuyume** [comment]
LGTM

**Comment by Hakuyume** [comment]
Sorry, this function doesn't assume the number of channels is three. It may work with grayscale images.
Could you add test cases of grayscale?

**Comment by yuyu2172** [comment]
Yeah. Thanks for pointing that out.

**Pull request merged**